### ISSUE TITLE
Weird graphical glitches caused by 2d batching 

### ISSUE THREAD

**Issue by MightyPrinny** [godotengine/godot#48981]
**Godot version:**
3.3.1 Mono

**OS/device including version:**
MXLinux 19/Debian 10 64bits, GLES2

**Issue description:**
I'm not sure what's going on exaclty, but I was able to replicate part of the glitches I was seeing in a project, 

https://user-images.githubusercontent.com/18746022/119241683-35299100-bb26-11eb-96bd-0a6d2105350d.mp4

Here things are disappearing, I was also seeing things being drawn at an offset in a project.

**Steps to reproduce:**
In the reproduction project I had to use different zindex, a script with custom 2d drawing and a grid lines shader.
It might be worth noting that the draw code uses global coordinates.
```
func _draw():
	if(texture == null):
		return;
	#global_position = referencePosition;
	draw_set_transform(-global_position,0,Vector2.ONE);
	
	var dir:Vector2 = (global_position-Vector2(0,16)-referencePosition).normalized();
	for i in range(length):
		draw_texture(texture,referencePosition+Vector2(-8,16)+i*16*dir);
```

**Minimal reproduction project:**
[DrawGlitch.zip](https://github.com/godotengine/godot/files/6527214/DrawGlitch.zip)

**Comment by lawnjelly** [comment]
Am just taking a look. I suspect there's something in the shader for the Node2D5 ColorRect that is causing a problem, just pinning it down.

**Comment by akien-mga** [comment]
Fixed by #48992.
